cpp/spglib.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <array>
+#include <exception>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+
+namespace Spglib {
+	/**
+	 * Get the version of the Spglib C++ binding
+	 *
+	 * @return X.Y.Z version
+	 */
+	const std::string version();
+	/**
+	 * Get the full version of the Spglib C++ binding
+	 *
+	 * @return full version
+	 */
+	const std::string version_full();
+	/**
+	 * Get the commit of the Spglib C++ binding
+	 *
+	 * @return commit
+	 */
+	const std::string commit();
+	/**
+	 * C++ exception object for spglib related errors
+	 */
+	class SpglibError : public std::exception {
+		const std::string errorMsg;
+
+	public:
+		const char* what() const noexcept override;
+		SpglibError(std::string&& message);
+	};
+	class Symmetry {
+		// TODO: Find better interface compatible with std::mdspan
+		std::array<double, 9> rotations;
+		std::array<double, 3> translations;
+	};
+	class MagneticSymmetry : public Symmetry {
+		bool time_reversal;
+	};
```

Comment:
Should this be split into `RotationSymmetry` and `TranslationSymmetry`, or are there cases when both are simultaneously non-zero? Similarly with `MagneticSymmetry` -> `TimeReversalSymmetry`?

And there's also `PointSymmetry` that is a subset of these.

Reply:
Time reversal operation (prime operation) commutes with space group operation, so this inheritance would work, but I am not very sure, and not also sure if this is intuitive. @lan496 can consider in depth on this.

Reply:
If this class was used in implementing core algorithms, the split or reconsideration would be needed. But I think this is thin wrapper for C-API, so the current one looks good.